validator/src/admin_rpc_service.rs

```diff
@@ -233,6 +233,74 @@ pub trait AdminRpc {
         meta: Self::Metadata,
         public_tpu_addr: SocketAddr,
     ) -> Result<()>;
+
+    #[rpc(meta, name = "setPublicTpuForwardsAddress")]
+    fn set_public_tpu_forwards_address(
+        &self,
+        meta: Self::Metadata,
+        public_tpu_forwards_addr: SocketAddr,
+    ) -> Result<()>;
+}
+
+macro_rules! set_public_address {
```

Comment:
Oh, I expected such a reaction :) I initially did it without macros. However, the resulting copy-paste was too extensive - about 50 lines, making it inconvenient to compare. So, I decided to switch to the macro version. But no problem, I'll show you how it would look without macros. I understand that the team's opinion carries more weight here than my personal preferences.